### PR TITLE
ISSD-1573 [CP 2.0] Added validation logic for multiple available keys

### PR DESCRIPTION
**Pull request by samuelleolopes**
[ISSD-78](https://liferay.atlassian.net/browse/ISSD-78)

### CODE REVIEW

**Comment by samuelleolopes** [comment]
## :fire: GitRay Report 
Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.
 ## Jira Tasks 

 <html></html> 

---

Hey, don't forget to update all related tickets with correct status, such as **In Review** :wink: 
### Git Stats 

1 file changed, 25 insertions(+), 4 deletions(-)

**Comment by samuelleolopes** [comment]
ci:test:sf

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/SelectSubscription.jsx [hunk 1]
```diff
@@ -52,6 +52,10 @@ const SelectSubscription = ({
 
 	const navigate = useNavigate();
 	const {state} = useLocation();
+	const [
+		totalOfActivationKeysAvailable,
+		setTotalOfActivationKeysAvailable,
```
Comment (kylembischof) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/SelectSubscription.jsx:57]: can we rename these `availableActivationKeysTotal` and `setAvailableActivationKeysTotal`, that will get rid of the `Of` in the name and make it a little shorter and precise.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/SelectSubscription.jsx [hunk 2]
```diff
@@ -189,14 +193,23 @@ const SelectSubscription = ({
 
 	const productName = [...new Set(productNames)].join(', ');
 
-	const inputDisplayName = productGroupName + ' ' + productName;
+	const nameSelectedProduct = state.activationKeys?.map((item) => {
```
Comment (kylembischof) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/SelectSubscription.jsx:196]: rename to `selectedProductName`

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/SelectSubscription.jsx [hunk 3]
```diff
@@ -189,14 +193,23 @@ const SelectSubscription = ({
...
+		return item.productName;
+	});
+
+	const uniqueNameSelectedProduct = [...new Set(nameSelectedProduct)].join(
```
Comment (kylembischof) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/SelectSubscription.jsx:200]: rename to `uniqueSelectedProductName`

**Review by kylembischof — COMMENTED** [review]
just a few small changes to make the variable naming more concise. other than that it looks good

**Comment by samuelleolopes** [comment]
ci:test:sf

**Comment by samuelleolopes** [comment]
ci:test:relevant

**Comment by samuelleolopes** [comment]
ci:test:sf

**Comment by samuelleolopes** [comment]
ci:test:relevant

**Comment by samuelleolopes** [comment]
ci:forward

**Comment by samuelleolopes** [comment]
ci:test:relevant

**Comment by samuelleolopes** [comment]
ci:forward

**Pull request closed without merging**